File: invoice-agent-system-java/src/main/java/com/example/invoice/config/InvoiceProperties.java
package com.example.invoice.config;

import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "invoice")
public record InvoiceProperties(
        Ai ai,
        Geneva geneva,
        ClientCallback clientCallback
) {
    public InvoiceProperties {
        ai = ai == null ? new Ai(null, null) : ai;
        geneva = geneva == null ? new Geneva(null, null, null) : geneva;
        clientCallback = clientCallback == null ? new ClientCallback(null, null, null) : clientCallback;
    }

    public record Ai(String mode, OpenAi openai) {
        public Ai {
            mode = mode == null || mode.isBlank() ? "multi" : mode;
            openai = openai == null ? new OpenAi(null, null, null, null) : openai;
        }
    }

    public record OpenAi(
            String apiKey,
            @NotBlank String baseUrl,
            String model,
            Long timeoutSeconds
    ) {
        public OpenAi {
            apiKey = apiKey == null ? "" : apiKey;
            baseUrl = baseUrl == null || baseUrl.isBlank() ? "https://api.openai.com" : baseUrl;
            model = model == null || model.isBlank() ? "gpt-4o-mini" : model;
            timeoutSeconds = timeoutSeconds == null ? 60L : timeoutSeconds;
        }
    }

    public record Geneva(
            @NotBlank String baseUrl,
            String apiKey,
            Long timeoutSeconds
    ) {
        public Geneva {
            baseUrl = baseUrl == null || baseUrl.isBlank() ? "http://localhost:9091" : baseUrl;
            apiKey = apiKey == null ? "" : apiKey;
            timeoutSeconds = timeoutSeconds == null ? 30L : timeoutSeconds;
        }
    }

    public record ClientCallback(
            Boolean enabled,
            String baseUrl,
            String apiKey
    ) {
        public ClientCallback {
            enabled = enabled != null && enabled;
            baseUrl = baseUrl == null || baseUrl.isBlank() ? "http://localhost:9092" : baseUrl;
            apiKey = apiKey == null ? "" : apiKey;
        }
    }
}

File: invoice-agent-system/src/main/kotlin/com/example/invoice/config/InvoiceProperties.kt
package com.example.invoice.config

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "invoice")
data class InvoiceProperties(
    val ai: Ai = Ai(),
    val geneva: Geneva = Geneva(),
    val clientCallback: ClientCallback = ClientCallback()
) {
    data class Ai(
        val mode: String = "multi",
        val openai: OpenAi = OpenAi()
    )

    data class OpenAi(
        val apiKey: String = "",
        @field:NotBlank val baseUrl: String = "https://api.openai.com",
        val model: String = "gpt-4o-mini",
        val timeoutSeconds: Long = 60
    )

    data class Geneva(
        @field:NotBlank val baseUrl: String = "http://localhost:9091",
        val apiKey: String = "",
        val timeoutSeconds: Long = 30
    )

    data class ClientCallback(
        val enabled: Boolean = false,
        val baseUrl: String = "http://localhost:9092",
        val apiKey: String = ""
    )
}
